src/c++/tests/cc_client_test.cc

```diff
@@ -1613,6 +1617,491 @@ TEST_F(GRPCTraceTest, GRPCClearTraceSettings)
       << std::endl;
 }
 
+class TestHttpInferRequest : public tc::HttpInferRequest {
```

Comment:
I think we need to break this test up a bit. Maybe label what you are testing with comments?